SDL2 HAVE_M_PI definition

Building the engine with make gives me error from multiple definitions from HAVE_M_PI.
File src/kre/SurfaceSDL.cpp gives this error. It has the definition with #ifndef, but before the include from SDL. File src/tiled/tiled.cpp algo gives the error. The definition is also before the include, in this case without the conditional definition.

For both files using #ifndef after the include solves this problem.

Edit: Im curious that no one else had this problem. Perhaps SDL library in my system should have some conditional definition that isn't there?

You are not the first one having this problem. From logs of the Discord server:

> <name_removed_0> - 06/29/2018
> I get an error installing the game on linux
> in making the anura engine I get
> /usr/include/SDL2/SDL_config.h:132: error: "HAVE_M_PI" redefined [-Werror]
>  \#define HAVE_M_PI 1
> src/kre/SurfaceSDL.cpp:28: note: this is the location of the previous definition
>  \# define HAVE_M_PI
> both from the AUR in arch linux, and building from soucer (cloning trunk branch and manually downloading imgui folder that doesn't get pulled by git)
> should I try to remove the Werrors flag from gcc? does anybody know?
>
> <name_removed_1> - 06/29/2018
> @<name_removed_0> maybe removing these three lines help you: [https://github.com/anura-engine/anura/blob/1c1ec65abb9900a7356cd7ef726d57412b08e561/src/kre/SurfaceSDL.cpp#L27-L29](https://github.com/anura-engine/anura/blob/1c1ec65abb9900a7356cd7ef726d57412b08e561/src/kre/SurfaceSDL.cpp#L27-L29), in any case, I dunno what will be the generic solution good for everybody.(edited)
> For some reason the compiler gets to define the symbol there before it is defined by your SDL.
> Alternatively, I guess you could make your SDL header to define that symbol only if it is not already defined (like [...] there in Anura). But maybe edit Anura better instead of SDL...
If you could report that at the issues tracker, maybe [...].
> 
> <name_removed_0> - 06/29/2018
> i'll report it later today thanks

Maybe this is a Makefile issue, if the compiler is supposed to scan all the inclusions before current project files. I don't know. At least it has a trivial workaround.

--- a/src/kre/SurfaceSDL.cpp
+++ b/src/kre/SurfaceSDL.cpp
@@ -24,9 +24,6 @@
 #ifndef _USE_MATH_DEFINES
 #	define _USE_MATH_DEFINES	1
 #endif 
-#ifndef HAVE_M_PI
-#	define HAVE_M_PI
-#endif 
 
 #include "SDL_image.h"
 

--- a/src/tiled/tiled.cpp
+++ b/src/tiled/tiled.cpp
@@ -21,7 +21,6 @@
 	   distribution.
 */
 
-#define HAVE_M_PI
 #include "SDL.h"
 #include "SDL_image.h"
 

